### PR TITLE
Allow more user details in the registration form

### LINKED ISSUE #1: System should input user details on registration
At the time of signup, input user full name (mandatory), age (mandatory), interest(dropdown i.e Events, Portraits, Weddings) not mandatory and save it into database.

### PR DESCRIPTION
**Pull request by sawleha16**
resolves #1 

1. Added custom attributes to users table (username:string, age:integer, interests:string)
2. Allowed new attributes in strong parameters [here](https://github.com/sawleha16/Photography-application/pull/4/files#diff-766c34fd6533171eaf54300c153f89d6002c35c02cfc9c5b219251f85180ad07)
3. Added those attributes in [view](https://github.com/sawleha16/Photography-application/pull/4/files#diff-744d99555c831425cf5883c199055bc537850fbdd075776c69dbad920497bb1f) as well

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::Base
+  before_action :confirgure_permitted_parameters,if: :devise_controller?
+
+  protected
+  def confirgure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) {|u| u.permit(:email,:password,:password_confirmation,:username,:age,:interests)}
+  end
```
Comment (ahmad19) [inline app/controllers/application_controller.rb]:
```suggestion
  protected
  
  def confirgure_permitted_parameters
    devise_parameter_sanitizer.permit(:sign_up) {|u| u.permit(:email, :password, :password_confirmation, :username, :age, :interests)}
  end
```

**Comment by ahmad19** [comment]
1. We need to have a dropdown in the registration form for interests
2. Add new attributes on the edit profile as well so user can edit their details

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (sawleha16) [reply app/controllers/application_controller.rb]: it changed

File: app/models/user.rb [hunk 2]
```diff
@@ -4,5 +4,6 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :confirmable, :lockable, :trackable
+         INTEREST_FIELD =["Wedding", "Events", "Potraits"]
```
Comment (ahmad19) [inline app/models/user.rb]: single quotes  and alignment :)

File: app/views/devise/registrations/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :age %><br />
+    <%= f.text_field :age, autocomplete: "age" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :interests%><br />
+    <%= f.select(:interests, User::INTEREST_FIELD) %>
```
Comment (ahmad19) [inline app/views/devise/registrations/edit.html.erb]: This should also have a blank by default.

File: app/models/user.rb (hunk 2 above)
Comment (ahmad19) [reply app/models/user.rb]: Add a presence validation for username and age.

File: app/models/user.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  validates :username, presence: true
+  validates :age, presence: true
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :confirmable, :lockable, :trackable
-
+         INTEREST_FIELD =['Wedding', 'Events', 'Potraits']
```
Comment (ahmad19) [inline app/models/user.rb]: Alignment

File: app/views/devise/registrations/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+  <div class="field">
+    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
+    <%= f.password_field :current_password, autocomplete: "current-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :username %><br />
+    <%= f.text_field :username, autocomplete: "username" %>
+    <%  resource.errors.full_messages_for(:username).each do |message| %>
```
Comment (ahmad19) [inline app/views/devise/registrations/edit.html.erb]: Remove these lines as it is already being handled by devise

File: app/views/devise/registrations/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    <div><%= message %></div>
+    <%end%>
+  </div>
+
+  <div class="field">
+    <%= f.label :age %><br />
+    <%= f.text_field :age, autocomplete: "age" %>
+    <%  resource.errors.full_messages_for(:age).each do |message| %>
```
Comment (ahmad19) [inline app/views/devise/registrations/edit.html.erb]: Remove these lines as it is already being handled by devise

File: app/views/devise/registrations/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
```
Comment (ahmad19) [inline app/views/devise/registrations/edit.html.erb]: Add turbo false

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :username %><br />
+    <%= f.text_field :username, autocomplete: "username" %>
+    <%  resource.errors.full_messages_for(:username).each do |message| %>
```
Comment (ahmad19) [inline app/views/devise/registrations/new.html.erb]: Remove these lines as it is already being handled by devise

File: app/views/devise/registrations/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    <div><%= message %></div>
+    <%end%>
+  </div>
+
+  <div class="field">
+    <%= f.label :age %><br />
+    <%= f.text_field :age, autocomplete: "age" %>
+    <%  resource.errors.full_messages_for(:age).each do |message| %>
```
Comment (ahmad19) [inline app/views/devise/registrations/new.html.erb]: Remove these lines as it is already being handled by devise

**Review by ahmad19 — APPROVED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (sawleha16) [reply app/models/user.rb]: Done

Comment (sawleha16) [reply app/models/user.rb]: > Add a presence validation for username and age.

Done

File: app/views/devise/registrations/edit.html.erb (hunk 3 above)
Comment (sawleha16) [reply app/views/devise/registrations/edit.html.erb]: > This should also have a blank by default.

done

File: app/views/devise/registrations/edit.html.erb (hunk 5 above)
Comment (sawleha16) [reply app/views/devise/registrations/edit.html.erb]: Done

File: app/views/devise/registrations/new.html.erb (hunk 9 above)
Comment (sawleha16) [reply app/views/devise/registrations/new.html.erb]: Done

File: app/views/devise/registrations/new.html.erb (hunk 8 above)
Comment (sawleha16) [reply app/views/devise/registrations/new.html.erb]: done

File: app/views/devise/registrations/edit.html.erb (hunk 7 above)
Comment (sawleha16) [reply app/views/devise/registrations/edit.html.erb]: Done

**Pull request merged**